server/sonar-server/src/test/java/org/sonar/server/computation/task/projectanalysis/component/ComponentImplTest.java

```diff
@@ -105,15 +105,15 @@ public void getFileAttributes_throws_ISE_if_BatchComponent_does_not_have_type_FI
 
   @Test
   public void isUnitTest_returns_true_if_IsTest_is_set_in_BatchComponent() {
-    ComponentImpl component = buildSimpleComponent(FILE, "file").setFileAttributes(new FileAttributes(true, null)).build();
+    ComponentImpl component = buildSimpleComponent(FILE, "file").setFileAttributes(new FileAttributes(true, null, 1)).build();
 
     assertThat(component.getFileAttributes().isUnitTest()).isTrue();
   }
 
   @Test
   public void isUnitTest_returns_value_of_language_of_BatchComponent() {
```

Comment:
A UT is missing on the IAE raised by constructor when the line parameter is < 0, and another one is missing on the value of the getLine().
That's debt, but there is also no UT for `FileAttributes#toString()` (you modified it, there is related change in UTs, it's like your change does not exist).